### PR TITLE
feat: rework --skip-sync to --sync with connectivity check

### PR DESCRIPTION
**Pull request by Cafe137**
Removed `--skip-sync`, added `--sync`, and `--sync` is now opt-in

---

When debug API is not available:

image

---

When there are no connected peers:

image

---

Prompts have been removed, only a warning message is displayed.

### CODE REVIEW

**Comment by AuHau** [comment]
LGTM, but I am wondering about the UX. Maybe I would be more proactive and just disabled the sync automatically without the prompt and only informed the user. The problem with this approach is that we would have to have a flag that could disable this behavior, which is not ideal. Although maybe we could use `--no-skip-sync`, I am not sure if the framework allows for bool flags to use `--no...` prefixes for `false` values.

I will leave the approval to @agazso as he is more aware of the UX side of swarm-cli.

**Comment by Cafe137** [comment]
One thing I do see as an issue is that we are using the `--quiet` flag for non-interactive, but it probably should have its own option, as verbosity level should not effect interactiveness.

File: src/command/upload.ts [hunk 1]
```diff
@@ -320,4 +319,32 @@ export class Upload extends RootCommand implements LeafCommand {
 
     return false
   }
+
+  private async handleSyncSupport(): Promise<void | never> {
+    if (this.quiet || !this.sync) {
```
Comment (agazso) [inline src/command/upload.ts:324]: This makes it not possible to turn on syncing in quiet mode?

Comment (Cafe137) [reply src/command/upload.ts:324]: This is only about printing the warnings. Sync will work regardless of verbosity level.

Comment (agazso) [reply src/command/upload.ts:324]: Then maybe it would be better to rename the function to `printWarningWhenNoConnectedPeers` or something more relevant.

**Review by nugaon — APPROVED** [review]
LGTM, althought I expected a successful run test also with `--sync`

**Pull request merged**